src/server/req_delete.c

```diff
@@ -394,6 +394,7 @@ req_deletejob(struct batch_request *preq)
 				parent->ji_ajtrk->tkm_tbl[offset].trk_substate =
```

Comment:
Nice catch, yes we need to update in this case too, will work on this. I will also automate a test case to cover these changes.